mwdb/web/src/components/UserProfile.js

```diff
@@ -80,51 +98,69 @@ class UserProfile extends Component {
                             <td>Last login</td>
                             <td><DateString date={this.state.profile.logged_on}/></td>
                         </tr>
+                        {this.props.userLogin === this.state.profile.login &&
                         <tr>
                             <td>Last password set</td>
                             <td><DateString date={this.state.profile.set_password_on}/></td>
                         </tr>
+                        }
+                        {(this.props.userLogin === this.state.profile.login && this.capabilities.length > 0) &&
+                        <tr>
+                            <td>
+                                Capabilities
+                            </td>
+                            <td>
+                                <ul className="table-ul">
+                                        {
+                                            this.capabilities.map(
+                                                c => <li>{capabilitiesList[c]} (inherited
+                                                    from: {this.inheritedFrom(c).join(", ")})</li>)
+                                        }
+                                </ul>
+                            </td>
+                        </tr>
+                        }
+                        {(this.props.userLogin === this.state.profile.login && this.groups.length > 0) &&
+                        <tr>
+                            <td>
+                                Groups
+                            </td>
+                            <td>
+                                <ul className="table-ul">
+                                        {
+                                             this.groups.map(g => <li>{g.name}</li>)
+                                        }
+                                </ul>
+                            </td>
+                        </tr>
+                        }
                         <tr>
                             <td style={{textAlign: 'left'}} colspan="3">
-                                    <button type="button" className="btn btn-success" 
+                                {this.props.userLogin === this.state.profile.login ?
+                                    <button type="button" className="btn btn-success"
                                     onClick={this.requestPasswordChange} disabled = {this.state.pressedRequestPassword}>
                                         Request new password
                                     </button>
+                                    :
+                                    <button type="button" className="btn btn-success">
+                                        <Link to={makeSearchLink("uploader", this.state.profile.login)} style={{color:'black'}}>
```

Comment:
You don't need `{{color: 'black'}}`, just swap the `<button>` and `<Link>` elements
```
<Link ...>
    <button ...>
```